packages/cli/src/amplify.ts

```diff
@@ -3,9 +3,41 @@
 import { hideBin } from 'yargs/helpers';
 import { createMainParser } from './main_parser_factory.js';
 import { attachUnhandledExceptionListeners } from './error_handler.js';
+import { extractSubCommands } from './extract_subcommands.js';
+import {
+  AmplifyFault,
+  PackageJsonReader,
+  UsageDataEmitterFactory,
+} from '@aws-amplify/platform-core';
+import { fileURLToPath } from 'node:url';
 
-attachUnhandledExceptionListeners();
+const packageJson = new PackageJsonReader().read(
+  fileURLToPath(new URL('../package.json', import.meta.url))
+);
+const libraryVersion = packageJson.version;
 
-const parser = createMainParser();
+if (libraryVersion == undefined) {
+  throw new AmplifyFault('UnknownVersionFault', {
+    message:
+      'Library version cannot be determined. Check the library installation',
+  });
+}
```

Comment:
should this be throw after we `attachUnhandledExceptionListeners` ?

Reply:
usage data emitter requires the library version to be initialized.